pdr_backend/contract/base_contract.py

```diff
@@ -1,20 +1,26 @@
 from abc import ABC
+from typing import Optional
 
 from enforce_typing import enforce_types
 from sapphirepy import wrapper
 
 
 @enforce_types
 class BaseContract(ABC):
-    def __init__(self, web3_pp, address: str, contract_name: str):
+    def __init__(
+        self, web3_pp, address: str, contract_name: str, pk_name: Optional[str] = None
```

Comment:
A contract already sees web3_pp. It shouldn't need more info about where the private key comes from. That is, it shouldn't need the private key name specified. (Doing so adds a lot of extra code, as this PR shows.)

web3_pp already manages a private key.

Let's ask "what's the simplest thing that could possibly work" for the goal of [#599](https://github.com/oceanprotocol/pdr-backend/issues/599): "Enable staking on both sides".

First, more background.

base_predictoor agent currently makes this call to submit predictions
```python
self.feed_contract.submit_prediction(predval, stake, target_slot, wait_for_receipt=True)
```

The simplest-thing is a new predictoor approach 4 agent, where we JIT create a second feed on the second pk. (We don't even need Web3PP to manage <1 keys!) 

Here's how it would submit predictions;
```python
# given: stake_amt_up, stake_amt_down

# set up feeds
feed_up = self.feed_contract # uses PRIVATE_KEY, the default
feed_down = copy.deepcopy(feed_up)
pk_down = os.getenv("PRIVATE_KEY2")
feed_down.config = feed_b.web3_pp._web3_config = Web3Config(feed_a.web3_pp.rpc_url, pk2)

# submit txs
feed_up.submit_prediction(True, stake_amt_up, target_slot, wait_for_receipt=True)
feed_down.submit_prediction(False, stake_amt_down, target_slot, wait_for_receipt=True)
```